PocketKit/Sources/Sync/PersistentContainer.swift

```diff
@@ -61,13 +64,43 @@ public class PersistentContainer: NSPersistentContainer {
         storeDescription.setOption(true as NSNumber, forKey: NSPersistentHistoryTrackingKey)
         storeDescription.setOption(true as NSNumber, forKey: NSMigratePersistentStoresAutomaticallyOption)
 
-        loadPersistentStores { storeDescription, error in
-            if let error = error as NSError? {
-                fatalError("Unresolved error \(error), \(error.userInfo)")
+        spotlightIndexer = CoreDataSpotlightDelegate(forStoreWith: storeDescription, coordinator: self.persistentStoreCoordinator)
+        spotlightIndexer?.startSpotlightIndexing()
+    }
+
+    /// Attempts to load the persistent container. If the load errors, the persistent container
+    /// is destroyed and rebuilt, performing a closure if a reset was necessary.
+    /// - Parameter onReset: Called when the persistent container required to be reset
+    /// - Note: If a reset was necessary, all previous on-disk data will be removed.
+    public func load(onReset: @escaping () -> Void) {
+        loadPersistentStores { [weak self] storeDescription, error in
+            guard let self else { return }
+            if error != nil {
```

Comment:
do we want to log to Sentry that we found an error and we are attempting the reset?

Reply:
Probably!